### PR TITLE
[BEAM-1282] Support @FinishBundle output in DoFnTester

### PR DESCRIPTION
**Pull request by rfevang**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [x] Make sure tests pass via `mvn clean verify`.
 - [x] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [x] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

**Comment by kennknowles** [comment]
R: @kennknowles

Thanks! Taking a look.

File: sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/DoFnTester.java [hunk 1]
```diff
@@ -546,11 +546,6 @@ private TestFinishBundleContext() {
       fn.super();
     }
 
-    private void throwUnsupportedOutputFromBundleMethods() {
```
Comment (kennknowles) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/DoFnTester.java]: It looks like you can also delete the one on line ~642

**Comment by kennknowles** [comment]
I notice you have left the checkbox for ICLA unchecked. This is a small contribution, so an ICLA isn't necessary, but if it is not a problem for you and/or you plan on contributing more, please do file one.

File: sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/DoFnTester.java (hunk 1 above)
Comment (rfevang) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/DoFnTester.java]: Done!

**Comment by rfevang** [comment]
Thanks for the review!

I'll look at the ICLA and sign it before my next contribution if I make another one, right now I've just addressed a couple of smaller things that caused issues in my pipeline.

**Comment by rfevang** [comment]
Filed the ICLA.

**Comment by rfevang** [comment]
Ping! It'd be nice to get this in for 2.1.0.

**Comment by kennknowles** [comment]
LGTM! Merging now.

**Comment by kennknowles** [comment]
Do you have a JIRA account? I forgot to assign https://issues.apache.org/jira/browse/BEAM-1282 to you, but I would still assign the resolved issue to you for the record.

**Comment by rfevang** [comment]
My JIRA username is 'rfevang'.

Thanks!

**Pull request closed without merging**